Druid/DRBytecodeGenerator.class.st

```diff
@@ -251,11 +258,20 @@ DRBytecodeGenerator >> visitReturn: aDRReturn [
 
 { #category : 'ir-to-target' }
 DRBytecodeGenerator >> visitStoreInstVar: aDRStoreInstanceVariable [ 
-
+	
 	aDRStoreInstanceVariable operand1 acceptVisitor: self.
-	builder pushLiteral: aDRStoreInstanceVariable operand2 value.
-	aDRStoreInstanceVariable operand3 acceptVisitor: self.
-	builder send: #instVarAt:put:
+	aDRStoreInstanceVariable operand1 class = DRLoadTemporaryVariable ifTrue: [ 
```

Comment:
And this condition is **repeated code**, what is something that we want to avoid.

The problem is a **missing abstraction**. I think we want a method 
```st
self isSelf: aDRStoreInstanceVariable operand1
```